sdk/include/compartment-macros.h

```diff
@@ -78,6 +98,44 @@
 #define MMIO_CAPABILITY(type, name)                                            \
 	MMIO_CAPABILITY_WITH_PERMISSIONS(type, name, true, true, false, false)
 
+/**
+ * Provide a capability of the type `type *` referring to the pre-shared object
+ * with `name` as its name.  This macro can be used only in code (it cannot be
+ * used to initialise a global).
+ *
+ * The last arguments specify the set of permissions that this capability
+ * holds.  Pre-shared objects are always global and without store local.  They
+ * may optionally omit additional permissions.
```

Comment:
I don't see any commentary about the necessary corresponding `xmake` bits.  There's a worked example down in `tests/` but I think some prose might be in order?

Reply:
Yes, once this is merged I'll update the programmers' guide.